TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1932,3 +1927,201 @@ def meters_to_pixels(distance):
+    POIs together, unioning their subway lines.
+
+    Finally, re-sort the features in case the merging has caused
+    the subway stations to be out-of-order.
+    """
+
+    assert source_layer, 'normalize_and_merge_duplicate_stations: missing source layer'
+
+    if zoom < start_zoom:
+        return None
+
+    # we probably don't want to do this at higher zooms (e.g: 17 &
+    # 18), even if there are a bunch of stations very close
+    # together.
+    if end_zoom is not None and zoom > end_zoom:
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    seen_stations = {}
+    new_features = []
+    for feature in layer['features']:
+        shape, props, fid = feature
+
+        kind = props.get('kind')
+        name = props.get('name')
+        if name is not None and kind == 'station':
+            # this should match station names where the name is
+            # followed by a ()-bracketed list of line names. this
+            # is common in NYC, and we want to normalise by
+            # stripping these off and using it to provide the
+            # list of lines if we haven't already got that info.
+            m = station_pattern.match(name)
+
+            subway_lines = props.get('subway_lines', [])
+
+            if m:
+                # if the lines aren't present or are empty
+                if not subway_lines:
+                    lines = m.group(2).split(',')
+                    subway_lines = [x.strip() for x in lines]
+                    props['subway_lines'] = subway_lines
+
+                # update name so that it doesn't contain all the
+                # lines.
+                name = m.group(1).strip()
+                props['name'] = name
+
+            seen_idx = seen_stations.get(name)
+            if seen_idx is None:
+                seen_stations[name] = len(new_features)
+
+                # ensure that subway lines is present and is of
+                # list type for when we append to it later if we
+                # find a duplicate.
+                props['subway_lines'] = subway_lines
+                new_features.append(feature)
+
+            else:
+                # get the properties and append this duplicate's
+                # subway lines to the list on the original
+                # feature.
+                seen_props = new_features[seen_idx][1]
+
+                # make sure lines are unique
+                seen_subway_lines = set(seen_props['subway_lines'])
+                subway_lines = set(subway_lines)
+                subway_lines.update(seen_subway_lines)
+
+                seen_props['subway_lines'] = list(subway_lines)
+
+        else:
+            # not a station, or name is missing - we can't
+            # de-dup these.
+            new_features.append(feature)
+
+    # need to re-sort: removing duplicates would have changed
+    # the number of lines for each station.
+    sort_pois(new_features, zoom)
```

Comment:
If the common case is not to have subway stations, it might be better to check that we've seen any at all before re-sorting.

```
if seen_stations:
    sort_pois(new_features, zoom)
```